app/components/billing/payment.js

```diff
@@ -53,6 +58,38 @@ export default Component.extend({
     }
   }).drop(),
 
+  // amount_off and price are in cents
+  discountedPrice: computed('couponResult.{amount_off,percent_off}', 'selectedPlan.price', function () {
+    const price = Math.floor(this.selectedPlan.price / 100);
+    if (this.couponResult && this.couponResult.amount_off) {
+      const amountOff = this.couponResult.amount_off;
+      const discountedPrice = price - Math.floor(amountOff / 100);
+      return `$${discountedPrice}`;
+    } else if (this.couponResult && this.couponResult.percent_off) {
+      const percentageOff = this.couponResult.percent_off;
+      const discountedPrice = price - (price * percentageOff) / 100;
+      return `$${discountedPrice.toFixed(2)}`;
+    } {
+      return `$${price}`;
+    }
+  }),
+
+  validateCoupon: task(function* () {
+    try {
+      const result = yield this.api.get(`/coupons/${this.coupon}`);
```

Comment:
Would it make sense to handle coupons with Ember Data (add a `Coupon` model, adapters/serializers as necessary)? It seems like it would simplify things a lot and would allow to encapsulate all the related logic inside model class. It could also have relation to `plan` if needed. What do you think?